recipe/install_nvcc.sh

```diff
@@ -91,18 +91,25 @@ export CMAKE_ARGS="\${CMAKE_ARGS}"
 ### /CMake configurations
 
 mkdir -p "\${CONDA_BUILD_SYSROOT}/lib"
+mkdir -p "\${CONDA_PREFIX}/lib/stubs"
 
 # Add \$(libcuda.so) shared object stub to the compiler sysroot.
 # Needed for things that want to link to \$(libcuda.so).
 # Stub is used to avoid getting driver code linked into binaries.
 
-# Make a backup of \$(libcuda.so) if it exists
-if [[ -f "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" ]]
+
+if [[ "\${CONDA_BUILD}" == 1 ]]
 then
-  LIBCUDA_SO_CONDA_NVCC_BACKUP="\${CONDA_BUILD_SYSROOT}/lib/libcuda.so-conda-nvcc-backup"
-  mv "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" "\${LIBCUDA_SO_CONDA_NVCC_BACKUP}"
+  # Make a backup of \$(libcuda.so) if it exists
+  if [[ -f "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" ]]
+  then
+    LIBCUDA_SO_CONDA_NVCC_BACKUP="\${CONDA_BUILD_SYSROOT}/lib/libcuda.so-conda-nvcc-backup"
+    mv "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" "\${LIBCUDA_SO_CONDA_NVCC_BACKUP}"
+  fi
+  ln -s "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so"
+else
+  ln -sf "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_PREFIX}/lib/stubs/libcuda.so"
```

Comment:
`lib/stubs` is specific to cmake. What about other build systems?

Reply:
I agree with @kkraus14  we should think of this as matching the upstream cuda toolkit layout for empty/stub libraries. 

Since the cuda toolkit provides a `stubs` subdirectory, CMake supports searching it when it can't find a real library first.

Reply:
The point of this whole package is to do away with cudatoolkit weirdness by making it standard for the compiler. (just like what other distros do)

Reply:
Well the standard behavior would be for this stub package to live in `lib/stubs` and for the actual driver library to live in the sysroot. Technically what we're doing now is wrong where we can end up in situations where the stub `libcuda.so` is a newer version than what's running on the system and symbols could technically fail to be found at library load time.

This means instead of what we're currently doing with the stub library that we'd need to detect if the system is actively running a driver and if it is to copy or symlink those driver libraries into the conda environment sysroot.

Reply:
The linking to the stub libraries comes from [this recommendation]( https://github.com/NVIDIA/nvidia-docker/wiki/Frequently-Asked-Questions#how-do-i-link-against-driver-apis-at-build-time-eg-libcudaso-or-libnvidia-mlso ). Also this comes up a few times in issues ( for example https://github.com/NVIDIA/nvidia-docker/issues/374#issuecomment-298097944 )

Reply:
> Well the standard behavior would be for this stub package to live in lib/stubs and for the actual driver library to live in the sysroot.

This is wrong. Actual driver library `libcuda.so.1` should live in ldconfig path where the dynamic loader can find it or in `$PREFIX/lib`.
It's fine to have `libcuda.so` in `$PREFIX/lib` since it's only used for linking and the SONAME is `libcuda.so.1` which is different from `libcuda.so`.

Reply:
If you still want to keep this in `lib/stubs`, then you should add a corresponding change to `LDFLAGS` to look at `lib/stubs`.

Reply:
Or set `LIBRARY_PATH` to `lib/stubs`.

Reply:
Sure. Would this work?

Guessing we don't want the `-Wl,-rpath,...` piece, but please let me know if I've missed something there

```suggestion
  ln -sf "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_PREFIX}/lib/stubs/libcuda.so"
  export LDFLAGS="${LDFLAGS} -L${CONDA_PREFIX}/lib/stubs"
```

Reply:
Yes, but you need to restore it in deactivate.sh

Reply:
That makes sense. How would this play with the `compilers`, which is doing some of this same saving itself?

Reply:
If you go with my other suggestion, this code path will only be when the `compilers` package is not used.

Reply:
> This is wrong. Actual driver library `libcuda.so.1` should live in ldconfig path where the dynamic loader can find it or in `$PREFIX/lib`.

In a normal installation both `libcuda.so` and `libcuda.so.1` come from the driver and get put in `/lib/x86_64-linux-gnu/` which I believe is the system sysroot (as opposed to the conda sysroot). The stub `libcuda.so` comes from the toolkit and ends up in the `stubs` directory.

Outside of conda, if someone is building something that uses the CUDA driver library without a driver installed on the machine, it is their responsibility to either point their compiler to the stub library or use something like CMake that automatically handles it for them. CMake will try to find a driver installed and running on the system first, and if it cannot find it will then fall back to using the stubs. Automatically setting `LDFLAGS` or `LIBRARY_PATH` feels like this package overstepping its boundaries.

> It's fine to have `libcuda.so` in `$PREFIX/lib` since it's only used for linking and the SONAME is `libcuda.so.1` which is different from `libcuda.so`.

I've seen plenty of code that dlopens `libcuda.so` (as opposed to `libcuda.so.1`) to sidestep manylinux violations and we would potentially blow that code up if we did this.

Reply:
Ah I see ( https://github.com/conda-forge/nvcc-feedstock/pull/84#discussion_r898285673 ). Yeah was thinking about this